src/components/vm/consoles/consoles.jsx

```diff
@@ -34,11 +43,70 @@ import './consoles.css';
 
 const _ = cockpit.gettext;
 
-const VmNotRunning = () => {
+const ConsoleEmptyState = ({ vm }) => {
+    const Dialogs = useDialogs();
+    const serials = vm.displays && vm.displays.filter(display => display.type == 'pty');
+    const inactive_vnc = vm.inactiveXML.displays && vm.inactiveXML.displays.find(display => display.type == 'vnc');
+
+    function add_vnc() {
+        Dialogs.show(<AddEditVNCModal idPrefix="add-vnc" vm={vm} consoleDetail={null} />);
+    }
+
+    function edit_vnc() {
+        Dialogs.show(<AddEditVNCModal idPrefix="edit-vnc" vm={vm} consoleDetail={inactive_vnc} />);
+    }
+
+    let icon = null;
+    let top_message = null;
+    let bot_message = _("Graphical console support not enabled");
+
+    if (serials.length == 0 && !inactive_vnc) {
+        bot_message = _("Console support not enabled");
+    } else {
+        if (vm.state != "running") {
+            if (serials.length > 0 && !inactive_vnc) {
+                top_message = _("Start the virtual machine to access its serial console");
+            } else {
+                top_message = _("Start the virtual machine to access its console");
+            }
+        } else if (inactive_vnc) {
+            icon = <EmptyStateIcon icon={PendingIcon} />;
+            if (serials.length > 0) {
+                top_message = _("Restart this virtual machine to access its graphical console");
+            } else {
+                top_message = _("Restart this virtual machine to access its console");
+            }
+        }
+    }
+
     return (
-        <div id="vm-not-running-message">
-            {_("Please start the virtual machine to access its console.")}
-        </div>
+        <EmptyState>
+            <EmptyStateHeader icon={icon} />
+            <EmptyStateBody>
+                {top_message}
+            </EmptyStateBody>
+            { !inactive_vnc
+                ? <>
+                    <EmptyStateBody>
+                        {bot_message}
+                    </EmptyStateBody>
+                    <EmptyStateFooter>
+                        <EmptyStateActions>
+                            <Button variant="secondary" onClick={add_vnc}>
+                                {_("Add VNC")}
+                            </Button>
+                        </EmptyStateActions>
+                    </EmptyStateFooter>
+                </>
+                : <EmptyStateFooter>
+                    <EmptyStateActions>
+                        <Button variant="link" onClick={edit_vnc}>
+                            {_("VNC server settings")}
```

Comment:
Similar language incongruence: What's "server" now? "Add VNC" → "VNC settings", or if  you match it up to the other one, "VNC console settings".

Reply:
Yes, agreed. I took this from the dialog title. I'll change both to just "VNC settings". @garrett, what's your opinion?

Reply:
I have also changed the dialog title "Add VNC server" to just "Add VNC".